pkg/kubecost/allocation.go

```diff
@@ -2224,6 +2227,10 @@ func (asr *AllocationSetRange) Length() int {
 
 // MarshalJSON JSON-encodes the range
 func (asr *AllocationSetRange) MarshalJSON() ([]byte, error) {
+	if asr == nil {
+		return json.Marshal([]*AllocationSet{})
```

Comment:
I'm not sure if this the "correct" behavior for a nil marshal...

@mbolt35 do you maybe have an opinion here?

Reply:
I think using an empty AllocationSet is probably fine, although I think `nil` should be supported, and it does depend on whatever decoder is receiving the JSON... There's an example of returning `"null"` in go:

https://github.com/golang/go/blob/bebe9aa42322f951fc3972c263648297bf9e04d4/src/math/big/intmarsh.go#L68-L74

Where if the receiver object is `nil` they just return `[]byte("null")` -- I am definitely ok with either of these solutions, maybe even partial to the empty array encode. As long as we know we are correctly processing the encoding on the frontend, we good 👍

Reply:
It seems reasonable to have an error here. In the past it seems that we have not allowed nil Sets and SetRanges, doing an error would continue to prohibit those without the panic, but we would still need to track down why we are getting nil asr now instead of empty ones.